### DISCUSSION TITLE
Issue with linting/formatting and the use of custom directives

### DISCUSSION THREAD

**Discussion by JaielZeus** [solidjs/solid#1872] [General]
[I have opened a question on SO about my issue](https://stackoverflow.com/questions/77048551/custom-directive-in-solidjs-is-undefined-and-gets-formatted-away) and trying to find further help here too.

I am unable to get custom directives to work properly. First my linter/formatter in VS code is separating the ``use:__`` directive at the colon. Secondly my custom directive ``use:model`` is reported as udnefined although I declared it as is explained in the docs.

Can anyone help me please?

**Comment by snnsnn** [comment]
I added an answer on SO: https://stackoverflow.com/a/77049172/7134134

**Reply by JaielZeus** [reply]
hey thank you for the answer. I am not sure it answer my question but I can accept it anyways. The tree shaking is correct this is what the docs also say. I had the undefined error in the editor already not in runtime but maybe that had to do with the formatting bug. I am more concerned about the formatting thing that keeps happening. I am using eslint with airbnb lint config and the solid plugin config. I am surely not the only one out there having this setup and trying to make use of custom directives. this is so strange and I cannot pinpoint it at all, so frustrating

**Reply by JaielZeus** [reply]
I deactivated every extension I have on vscode and it inserts a space after the colon without any extension being involved so eslint is not it. I am gonna open a new issue about it on SO I think
